librepo/downloader.c

```diff
@@ -1303,6 +1303,9 @@ check_zck(LrTarget *target, GError **err)
     assert(!err || *err == NULL);
     assert(target && target->f && target->target);
 
+    /* Reset range fail flag */
+    target->range_fail = FALSE;
+
```

Comment:
I'd like to take a closer look at the logic for this commit, since it's not clear to me what this is changing.

Reply:
Ok, I've looked more closely at this and I think it makes sense.  I've also rebuilt `librepo` with these commits added and can verify that it doesn't seem to break anything (I tested against an FTP mirror and a mirror without range support)

I am a bit concerned that I wasn't able to reproduce the bug https://bugzilla.redhat.com/show_bug.cgi?id=1886706 on my F33 system, either with stock librepo-1.12.1 or with this PR applied, so I'm not sure whether this PR actually fixes the problem.

Reply:
Thanks for looking into this. I have actually changed it a bit more and added a new commit. I have tried to explain it in the commit messages but since its quite complex I am not sure they are readable.

That is weird I was able to reproduce it quite easily. Just having a zchunk repo served over FTP was enough. I have created tests for out ci-dnf-stack: https://github.com/rpm-software-management/ci-dnf-stack/pull/903 which also reproduce it (specifically the scenario with the `@bz1886706` tag).

Reply:
@jdieter would you please review the latest changes? They are quite involved indeed...